Core/include/Acts/Propagator/TryAllNavigator.hpp

```diff
@@ -576,58 +578,24 @@ class TryAllOverstepNavigator : public TryAllNavigatorBase {
                                                Logging::INFO))
       : TryAllNavigatorBase(std::move(cfg), std::move(logger)) {}
 
-  State makeState(const Surface* startSurface,
-                  const Surface* targetSurface) const {
-    State result;
-    result.startSurface = startSurface;
-    result.targetSurface = targetSurface;
-    return result;
-  }
-
-  const Surface* currentSurface(const State& state) const {
-    return state.currentSurface;
-  }
-
-  const TrackingVolume* currentVolume(const State& state) const {
-    return state.currentVolume;
-  }
-
-  const IVolumeMaterial* currentVolumeMaterial(const State& state) const {
-    if (state.currentVolume == nullptr) {
-      return nullptr;
-    }
-    return state.currentVolume->volumeMaterial();
-  }
-
-  const Surface* startSurface(const State& state) const {
-    return state.startSurface;
-  }
-
-  const Surface* targetSurface(const State& state) const {
-    return state.targetSurface;
-  }
-
-  bool targetReached(const State& state) const { return state.targetReached; }
-
-  bool endOfWorldReached(State& state) const {
-    return state.currentVolume == nullptr;
-  }
-
-  bool navigationBreak(const State& state) const {
-    return state.navigationBreak;
-  }
-
-  void currentSurface(State& state, const Surface* surface) const {
-    state.currentSurface = surface;
-  }
+  State makeState(const Options& options) const {
+    assert(options.startSurface != nullptr && "Start surface must be set");
 
-  void targetReached(State& state, bool targetReached) const {
-    state.targetReached = targetReached;
+    State state;
+    state.options = options;
+    state.startSurface = options.startSurface;
+    state.targetSurface = options.targetSurface;
+    return state;
   }
 
-  void navigationBreak(State& state, bool navigationBreak) const {
-    state.navigationBreak = navigationBreak;
-  }
+  using TryAllNavigatorBase::currentSurface;
+  using TryAllNavigatorBase::currentVolume;
+  using TryAllNavigatorBase::currentVolumeMaterial;
+  using TryAllNavigatorBase::endOfWorldReached;
+  using TryAllNavigatorBase::navigationBreak;
+  using TryAllNavigatorBase::startSurface;
+  using TryAllNavigatorBase::targetReached;
+  using TryAllNavigatorBase::targetSurface;
```

Comment:
This is a bit ugly. I guess with a concept we could ensure the navigator types have all of these in a central place.

Reply:
Yeah I also don't like it but I don't see an alternative without using virtual inheritance